src/main/java/seedu/address/model/person/Location.java

```diff
@@ -0,0 +1,52 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+public class Location {
+
+    public static final String MESSAGE_CONSTRAINTS = "Locations can take any values, and it should not be blank";
+
+    /*
+     * The first character of the location must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "[^\\s].*";
+
+    public final String value;
+
+    /**
+     * Constructs a Location.
+     *
+     * @param location A valid location.
+     */
+    public Location(String location) {
+        requireNonNull(location);
```

Comment:
Location should be a necessary field for professors an TAs as we can know where they are conducting their lessons at. However, location should not be necessary field for students.

Reply:
I can see where you're coming from, but I am not sure if I agree with that fully.
I think there could be useful user stories where one might want to know where one usually meets with another student.

E.g. One could meet a student at their Chess Club activities, and might want to indicate that in their contact information. Or even put their place of residence at campus in the contact information.

Reply:
Leaving location as an optional field might just solve this issue!